### PR TITLE
UI: Avoid getting into infinite render loop in CategoriesTree

### PR DESCRIPTION
**Pull request by grigasp**

### CODE REVIEW

File: ui/framework/src/ui-framework/imodel-components/category-tree/CategoryVisibilityHandler.tsx [hunk 1]
```diff
@@ -7,11 +7,13 @@
...
 
 import * as React from "react";
+import { BeEvent } from "@bentley/bentleyjs-core";
 import { IModelConnection, PerModelCategoryVisibility, ViewManager, Viewport } from "@bentley/imodeljs-frontend";
 import { NodeKey } from "@bentley/presentation-common";
 import { TreeNodeItem, useAsyncValue } from "@bentley/ui-components";
 import { IVisibilityHandler, VisibilityChangeListener, VisibilityStatus } from "../VisibilityTreeEventHandler";
-import { BeEvent } from "@bentley/bentleyjs-core";
+
+const EMPTY_CATEGORIES_ARRAY: Category[] = [];
```
Comment (saskliutas) [inline ui/framework/src/ui-framework/imodel-components/category-tree/CategoryVisibilityHandler.tsx:16]: Why is this needed? Maybe it would be enough to set correct return type on `useCategories` and `loadCategoriesFromViewport`?

Comment (grigasp) [reply ui/framework/src/ui-framework/imodel-components/category-tree/CategoryVisibilityHandler.tsx:16]: It's not for the type. 

The `useCategories` hook was returning a new empty array very time it was called. The `CategoryTree` component passes the array as a dependency for `useCategoryVisibilityHandler`, so a new visibility handler was getting created. Which in turn caused a re-render, which created a new empty categories array, new visibility handler, and so on.

**Review by saskliutas — APPROVED** [review]

**Review by bsteinbk — APPROVED** [review]

**Review by DanEastBentley — APPROVED** [review]

**Comment by grigasp** [comment]
@mergifyio backport release/2.15.x

**Comment by grigasp** [comment]
@mergifyio backport release/2.14.x
